### PR TITLE
[macOS] Remove usages of FlutterViewController#loadView

### PR DESCRIPTION
**Pull request by dkwingsmt**
The macOS specification clearly said that `loadView` should not be directly called, and that calling `view` when the view is nil will automatically trigger `loadView`. This PR removed almost all usages of `loadView` except for a few in tests. The PR also changes `FlutterViewController#flutterView`, which is equivalent to `view`, to also follow the convention of initializatio-on-first-use. 

The most important outcome is removing the `loadView` call in `FlutterViewController`'s initializer, as this extra call introduced in https://github.com/flutter/engine/pull/36410 confusingly complicates the initialization process.

cc @sergiy-sc Any reason you added this call? Was it a mistake?

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/macos/framework/Source/FlutterViewController.mm [hunk 1]
```diff
@@ -410,6 +409,16 @@ - (void)dealloc {
 
 #pragma mark - Public methods
 
+- (FlutterView*)flutterView {
+  if (!self.viewLoaded) {
```
Comment (jmagman) [inline shell/platform/darwin/macos/framework/Source/FlutterViewController.mm:413]: Minor nit:
```suggestion
  if (!self.isViewLoaded) {
```

On iOS there's `-[UIViewController viewIfLoaded]` but there doesn't seem to be an AppKit equivalent.

**Comment by sergiy-sc** [comment]
I used engine from master, and created a macos app. Replaced
```
let flutterViewController = FlutterViewController.init()
```

in the `MainFlutterWindow` (Runner.xcworkspace) with 
```
let flutterEngine = FlutterEngine(name: "Custom Engine", project: nil, allowHeadlessExecution: true)
flutterEngine.run(withEntrypoint: nil)
let flutterViewController = FlutterViewController.init(engine: flutterEngine, nibName: nil, bundle: nil)
```
If I remove `[self loadView]` it works fine (so looks like it's not needed). However, when I use @dkwingsmt 's patch from this PR, I get an exception:
```
2023-01-12 13:36:07.362984+0200 calculator[44134:56356473] Metal API Validation Enabled
flutter: The Dart VM service is listening on http://127.0.0.1:57525/9_nypM3niOA=/
=================================================================
Main Thread Checker: UI API called on a background thread: -[NSViewController isViewLoaded]
PID: 44134, TID: 56357358, Thread name: io.flutter.raster, Queue name: com.apple.root.default-qos.overcommit, QoS: 0
Backtrace:
4   FlutterMacOS                        0x000000010e3d3774 -[FlutterViewController flutterView] + 52
5   FlutterMacOS                        0x000000010e3d8540 -[FlutterViewEngineProvider getView:] + 124
6   FlutterMacOS                        0x000000010e3bb3b4 -[FlutterRenderer createTextureForView:size:] + 84
7   FlutterMacOS                        0x000000010e3bb06c _ZL31OnGetNextDrawableForDefaultViewP13FlutterEnginePK16FlutterFrameInfo + 240
8   FlutterMacOS                        0x00000001112bb29c _ZZL38InferMetalPlatformViewCreationCallbackPK21FlutterRendererConfigPvRKN7flutter20PlatformViewEmbedder21PlatformDispatchTableENSt21_LIBCPP_ABI_NAMESPACE10unique_ptrINS3_28EmbedderExternalViewEmbedderENS8_14default_deleteISA_EEEEENK5$_121clERK7SkISize + 128
9   FlutterMacOS                        0x00000001112bb1ec _ZNSt21_LIBCPP_ABI_NAMESPACE8__invokeB6v15000IRZL38InferMetalPlatformViewCreationCallbackPK21FlutterRendererConfigPvRKN7flutter20PlatformViewEmbedder21PlatformDispatchTableENS_10unique_ptrINS5_28EmbedderExternalViewEmbedderENS_14default_deleteISB_EEEEE5$_121JRK7SkISizeEEEDTclclsr3stdE7declvalIT_EEspclsr3stdE7declvalIT0_EEEEOSK_DpOSL_ + 48
10  FlutterMacOS                        0x00000001112bb12c _ZNSt21_LIBCPP_ABI_NAMESPACE28__invoke_void_return_wrapperIN7flutter17GPUMTLTextureInfoELb0EE6__callIJRZL38InferMetalPlatformViewCreationCallbackPK21FlutterRendererConfigPvRKNS1_20PlatformViewEmbedder21PlatformDispatchTableENS_10unique_ptrINS1_28EmbedderExternalViewEmbedderENS_14default_deleteISE_EEEEE5$_121RK7SkISizeEEES2_DpOT_ + 48
```

If I use `flutter run...` it works ok, but if I launch Runner.xcworkspace it shows that exception (doesn't matter if I create flutter engine myself or not). I reverted the patch - worked, I deleted the revert - got exception. I will do some more testing about `[self loadView]` but I think this patch needs a bit more work.

**Comment by dkwingsmt** [comment]
Thank you so much. I only ran against all unit tests. I'll take a look at the case you found.

**Comment by dkwingsmt** [comment]
Closing this PR for now.

**Pull request closed without merging**